### PR TITLE
Update CloudFoundry CLI to v7

### LINKED ISSUE #910: Add zero-downtime deployment capability to CloudFoundry deployments
With some care, it is possible to roll out CF app deployments with zero downtime. We should add such a capability to our deployment provider.

### PR DESCRIPTION
**Pull request by abdonrd**
Official instructions:

https://github.com/cloudfoundry/cli/blob/master/doc/installation-instructions/installation-instructions-v7.md#installers-and-compressed-binaries

Also adding support to the new CLI v7 has a new strategy option for rolling deploy:
https://docs.cloudfoundry.org/devguide/deploy-apps/rolling-deploy.html

Fix #910

### CODE REVIEW

File: lib/dpl/providers/cloudfoundry.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ class Cloudfoundry < Provider
       opt '--v3',                  'Use the v3 API version to push the application'
       opt '--logout', default: true, internal: true
 
-      cmds install: 'test $(uname) = "Linux" && rel="linux64-binary" || rel="macosx64"; wget "https://cli.run.pivotal.io/stable?release=${rel}&source=github" -qO cf.tgz && tar -zxvf cf.tgz && rm cf.tgz',
+      cmds install: 'test $(uname) = "Linux" && rel="linux64-binary" || rel="macosx64-binary"; curl -L "https://packages.cloudfoundry.org/stable?release=${rel}&version=v7&source=github" | tar -zx && mv cf /usr/local/bin',
```
Comment (BanzaiMan) [inline lib/dpl/providers/cloudfoundry.rb:28]: It just occurred to me:

https://packages.cloudfoundry.org/stable, without query parameters, warns us:

```
Invalid 'release' value, please select one of the following releases: debian32, debian64, redhat32, redhat64, macosx64, windows32, windows64, linux32-binary, linux64-binary, macosx64-binary, windows32-exe, windows64-exe
```

I get the feeling that `linux64-binary` would not work with ARM CPUs (which means we should test `$(uname -m)`), and we probably want to block Windows, too.

Neither of these things are show stopper, as they are not universally enforced at the moment, but I thought I'd point that out.

**Comment by BanzaiMan** [comment]
Specs need fixing.

File: spec/dpl/providers/cloudfoundry_spec.rb [hunk 2]
```diff
@@ -27,6 +27,10 @@
     it { should have_run './cf push -f other.yml' }
   end
 
+  describe 'given --strategy rolling' do
+    it { should have_run './cf push --strategy rolling' }
+  end
+
```
Comment (abdonrd) [inline spec/dpl/providers/cloudfoundry_spec.rb:33]: @BanzaiMan do you know why it fails? Thanks!

File: lib/dpl/providers/cloudfoundry.rb (hunk 1 above)
Comment (abdonrd) [reply lib/dpl/providers/cloudfoundry.rb]: Ok! For now I keep the current installation.

File: spec/dpl/providers/cloudfoundry_spec.rb (hunk 2 above)
Comment (BanzaiMan) [reply spec/dpl/providers/cloudfoundry_spec.rb:33]: I think `--strategy` is reserved to vary provider implementation (see `pages` and `heroku`). We'll need a different name.

Comment (abdonrd) [reply spec/dpl/providers/cloudfoundry_spec.rb]: I have no idea if the Cloudfoundry CLI will support more options in the future...

Comment (abdonrd) [reply spec/dpl/providers/cloudfoundry_spec.rb]: Do you prefer `deployment_strategy` or `rolling`?

**Comment by abdonrd** [comment]
@BanzaiMan it should works now! 😁

**Comment by BanzaiMan** [comment]
Great. Now, could you test this code to deploy your app?

**Comment by abdonrd** [comment]
@BanzaiMan ok! How can I test it?

**Comment by BanzaiMan** [comment]
It's explained in https://github.com/travis-ci/dpl#testing-dpl-branches-or-forks-on-travis-ci

**Comment by abdonrd** [comment]
@BanzaiMan yay! It's working!

https://travis.ibm.com/ETX/rendertron/builds/37482611

Screenshot 2020-09-02 at 17 51 29

Screenshot 2020-09-02 at 17 51 00

**Comment by abdonrd** [comment]
Friendly ping to @BanzaiMan. 😄

**Comment by Tansito** [comment]
Interested on this too! 👀

**Comment by abdonrd** [comment]
@BanzaiMan @eugene-travis when we can start to use this? Thanks in advance!

**Comment by BanzaiMan** [comment]
This should be available in the `--pre` release.

https://github.com/travis-ci/dpl/blob/v2.0.2.beta.1/lib/dpl/providers/cloudfoundry.rb#L25

**Comment by abdonrd** [comment]
Thanks @BanzaiMan!

To use the `--pre` I need to setup the `edge: true`, right?

Because with this:

Screenshot 2020-10-15 at 18 06 12

Now the provider doesn't work:

Screenshot 2020-10-15 at 18 06 38

**Comment by BanzaiMan** [comment]
`hpc` is not a string that should happen in normal processing. I tend to think that you have an incorrect value in `$IBMCLOUD_API_KEY` that leads to this invocation.

**Comment by abdonrd** [comment]
@BanzaiMan mmm but this deploy is working just without these changes:

Screenshot 2020-10-15 at 18 06 12

**Comment by BanzaiMan** [comment]
Hmmm. That is bizarre. Earlier, in https://github.com/travis-ci/dpl/pull/1211#issuecomment-685828183 you indicated that the code was working. :-( I am not quite sure what's happening.

**Comment by abdonrd** [comment]
@BanzaiMan yes, that's why I don't understand what is happening 😕

**Comment by abdonrd** [comment]
@BanzaiMan proposal: https://github.com/travis-ci/dpl/pull/1219

**Pull request merged**